assets/src/edit-story/components/panels/stylePreset/presets.js

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled, { css } from 'styled-components';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { ReactComponent as Remove } from '../../../icons/remove.svg';
+import { BACKGROUND_TEXT_MODE } from '../../../constants';
+import generatePatternStyles from '../../../utils/generatePatternStyles';
+import { PanelContent } from '../panel';
+import PresetGroup from './presetGroup';
+import { generatePresetStyle } from './utils';
+
+const REMOVE_ICON_SIZE = 18;
+
+const presetCSS = css`
+  display: inline-block;
+  width: 30px;
+  height: 30px;
+  border-radius: 15px;
+  border-color: transparent;
+  padding: 0;
+  font-size: 13px;
+  position: relative;
+  svg {
+    width: ${REMOVE_ICON_SIZE}px;
+    height: ${REMOVE_ICON_SIZE}px;
+    position: absolute;
+    top: calc(50% - ${REMOVE_ICON_SIZE / 2}px);
+    left: calc(50% - ${REMOVE_ICON_SIZE / 2}px);
+  }
+`;
+
+const Color = styled.button`
+  ${presetCSS}
+  ${({ color }) => generatePatternStyles(color)}
+`;
+
+const Style = styled.button`
+  ${presetCSS}
+  background: transparent;
+  ${({ styles }) => styles}
+  width: 72px;
+  border-radius: 4px;
+`;
+
+const TextWrapper = styled.span`
+  text-align: left;
+  line-height: 1.3;
+`;
+
+const HighLight = styled.span`
+  padding: 0 2px;
+  ${({ background }) => generatePatternStyles(background)}
+  box-decoration-break: clone;
+`;
+
+function Presets({
+  stylePresets,
+  getEventHandlers,
+  isEditMode,
+  isText,
+  textContent = 'Text',
+}) {
+  const { fillColors, textColors, textStyles } = stylePresets;
+
+  const getStylePresetText = (preset) => {
+    const isHighLight =
+      preset.backgroundTextMode === BACKGROUND_TEXT_MODE.HIGHLIGHT;
+    // @todo Confirm text content usage.
+    const text = textContent.substring(0, 8) + '...';
```

Comment:
Shouldn't this be done on CSS?

Reply:
Makes sense. I was originally thinking of keeping all the same length (e.g. in case of font-family differences) but that's not really necessary. Changed now.